var/spack/repos/builtin/packages/icedtea/package.py

```diff
@@ -0,0 +1,163 @@
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+import shutil
+import inspect
+
+class Icedtea(AutotoolsPackage):
+    """The IcedTea project provides a harness to build the source code from
+    http://openjdk.java.net using Free Software build tools and adds a number
+    of key features to the upstream OpenJDK codebase. IcedTea requires an
+    existing IcedTea or OpenJDK install to build."""
+
+    homepage = "http://icedtea.classpath.org/wiki/Main_Page"
+    url      = "http://icedtea.wildebeest.org/download/source/icedtea-3.4.0.tar.gz"
+
+    version('3.4.0',  'eba66765b92794495e16b83f23640872')
+
+    provides('java@8', when='@3.4.0:3.99.99')
+    variant('X', default=False, description="Build with GUI support.")
+    variant('shenandoah', default=False, description="Build with the shenandoah gc. Only for version 3+")
+
+    depends_on('pkg-config', type='build')
+    depends_on('gmake', type='build')
+    depends_on('cups')
+    #X11 deps required for building even when headless
+    depends_on('libx11', when='~X', type='build')
+    depends_on('xproto', when='~X', type='build')
+    depends_on('libxext', when='~X', type='build')
+    depends_on('libxtst', when='~X', type='build')
+    depends_on('libxi', when='~X', type='build')
+    depends_on('libxt', when='~X', type='build')
+    depends_on('libxinerama', when='~X', type='build')
+    depends_on('libxrender', when='~X', type='build')
+    depends_on('libxcomposite', when='~X', type='build')
+    depends_on('libxau', when='~X', type='build')
+    depends_on('libxdmcp', when='~X', type='build')
+    depends_on('gtkplus+X', when='~X', type='build')
+    
+    depends_on('libx11', when='+X')
+    depends_on('xproto', when='+X')
+    depends_on('libxext', when='+X')
+    depends_on('libxtst', when='+X')
+    depends_on('libxi', when='+X')
+    depends_on('libxt', when='+X')
+    depends_on('libxinerama', when='+X')
+    depends_on('libxrender', when='+X')
+    depends_on('libxcomposite', when='+X')
+    depends_on('libxau', when='+X')
+    depends_on('libxdmcp', when='+X')
+    depends_on('gtkplus+X', when='+X')
+
+    depends_on('freetype@2:')
+    depends_on('wget', type='build')
+    depends_on('giflib')
+    depends_on('libpng')
+    depends_on('jpeg')
+    depends_on('lcms')
+    depends_on('zlib')
+    depends_on('alsa-lib')
```

Comment:
So you're saying that in order to build `icedtea`, you need a pre-existing working Java installation, and that should already come with `jar` or `fastjar`? That makes sense. Honestly, the only reason we started working on `icedtea` in #2624 was because `jdk` has weird licensing and we wanted a way to bootstrap `openjdk` from source, but it turns out `icedtea` needs to be bootstrapped too.